library/strings.cpp

```diff
@@ -369,6 +402,19 @@ UString::ConstIterator UString::ConstIterator::operator++()
 	return *this;
 }
 
+UString::ConstIterator UString::ConstIterator::operator--()
+{
+	const char *ptr = s.cStr();
+	ptr += --this->offset;
+	while (static_cast<unsigned char>(*ptr) >= 0x10000000 &&
```

Comment:
This looks broken - an unsigned char can never be >= 0x1000000?

One cool thing about utf8 is you can find the start byte of all character as they're the only ones with the highest bit unset.

So this can be simplified to:
while (*ptr & 0x80) { go back 1 byte }
?

Reply:
You're right, it should have been [0**b**10000000 ; 0**b**11000000), which is the range for continuation bytes. Unfortunately the first byte in multi-byte characters have their highest bit set (unlike single-byte chars), but they are always outside the aforemetioned range

Reply:
Yeah, brain fart - I mean bit 2 (0b01000000) is only set on the first of a multibyte sequence. I somehow neglected single-byte chars, and got the wrong bit on top :)

So the first byte either has (byte & 0b10000000) for a single byte sequence, or (byte & 0b01000000) for the first of a multibyte. but I guess that's equivalent to the numerical comparisons)